src/model.jl

```diff
@@ -615,31 +718,173 @@ function read_sea_ice_extent_N(year=1979)
+end
+
 
 """
-calc_geography_per_month_extent!()
-* This routine a new geography matrix by determining sea ice extent with a monthly rate of change
+calculate_new_extent_N()
+* This routine computes a new geography matrix by determining sea ice extent with a monthly rate of change
   and mapping extent or reduction to adjacent sea ice cells and ocean cells on the Northern Hemisphere.
 
-* This routine assumes that the sea ice extent of the geography file corresponds to 
-  the measured sea ice at the vernal equinox.
- 
 * There is a bias in the order sea ice cells are added/removed.
   sea ice extent: starts from top left to bottom right
   sea ice reduction: starts from bottom left to top right
+"""
+function calculate_new_extent_N(geography,new_extent,area_frac, nlongitude=128, nlatitude =65)
+
+  nlat::Int64 = (nlatitude-1)*0.5
+
+  G = geography[:,1:nlat]
+  N = reverse(G, dims = 2)
+  IndG = CartesianIndices(G)
+  IndN = CartesianIndices(N)
+  Gfirst, Glast = first(IndG), last(IndG)
+  Ifirst, Ilast = first(IndN), last(IndN)
+  G1 = oneunit(Gfirst)
+  I1 = oneunit(Ifirst)
+
+  extent_remaining = new_extent
+ 
+  # case for a sea ice extent
+  if extent_remaining > 0
+    for I in IndG
+        if G[I[1], I[2]] == 2 || G[I[1], I[2]] == 3
+            if extent_remaining <= 0
+                break
+            else
+                for J in max(Gfirst, I-G1):min(Glast, I+G1)
+                    if G[J] in (5,6,7,8) && extent_remaining > 0
+                        G[J[1], J[2]] = 2
+                        extent_remaining -= area_frac[J[2]]
+                    end
+                end
+            end
+        end
+    end
+    geography[:,1:nlat] = G
+
+ # case for a reduction of sea ice
+ # adjacent sea ice cells will turn to ocean cells
+ elseif extent_remaining < 0
+    for I in IndN
+        if N[I[1], I[2]] == 2 && extent_remaining < 0
+            N[I[1], I[2]] = 5
+            extent_remaining += reverse(area_frac[1:nlat])[I[2]]
+            if extent_remaining >= 0
+                break
+            else
+                for J in max(Ifirst, I-I1):min(Ilast, I+I1)
+                    if N[J] == 2 && extent_remaining < 0
+                        N[J[1], J[2]] = 5
+                        extent_remaining += reverse(area_frac[1:nlat])[J[2]]
+                    end
+                end
+
+            end  
+            
+        end
+    end 
+    geography[:,1:nlat] = reverse(N, dims = 2)
+ end
+
+  return geography
+end
+
 
 """
-function calc_geography_per_month_extent!(geography,annual_sea_ice_extent, time_step, nlongitude=128, nlatitude =65)
+calculate_new_extent_S()
+* This routine computes a new geography matrix by determining sea ice extent with a monthly rate of change
+  and mapping extent or reduction to adjacent sea ice cells and ocean cells on the Southern Hemisphere.
 
-  month::Int64 = 0
+* There is a bias in the order sea ice cells are added/removed.
+  sea ice extent: starts from bottom left to top right
```

Comment:
Please check if the extent is from bottom left to top right or from bottom right to top left